Libraries/LibJS/Runtime/ArrayPrototype.cpp

```diff
@@ -229,4 +230,33 @@ Value ArrayPrototype::join(Interpreter& interpreter)
     return join_array_with_separator(interpreter, *array, separator);
 }
 
+Value ArrayPrototype::concat(Interpreter& interpreter)
+{
+    auto* array = array_from(interpreter);
+    if (!array)
+        return {};
+
+    auto* new_array = interpreter.heap().allocate<Array>();
+    auto& start_array = *array->elements().begin();
+    auto* start_array_values = const_cast<const Value*>(&start_array);
+    new_array->elements().append(start_array_values, array->elements().size());
```

Comment:
Can we just do this with `Vector::append(const Vector&)`?:

```suggestion
    new_array->elements().append(array->elements());
```

Reply:
same case as the previous